swaylock/password.c

```diff
@@ -97,7 +98,10 @@ void swaylock_handle_key(struct swaylock_state *state,
 				state->eventloop, 50, handle_preverify_timeout, state);
 
 		while (state->run_display && state->verify_password_timer) {
-			wl_display_flush(state->display);
+			errno = 0;
+			if (wl_display_flush(state->display) == -1 && errno != EAGAIN) {
```

Comment:
I'm not sure whether checking for EAGAIN is relevant here. If we do, we should probably do it in a loop. But w/e.

Reply:
If by loop, you mean, keep running `wl_display_flush`, the documentation says: "wl_display_flush() never blocks. It will write as much data as possible, but if all data could not be written, errno will be set to EAGAIN and -1 returned. In that case, use poll on the display file descriptor to wait for it to become writable again." so you can't do that.

Reply:
```c
errno = 0;
while (wl_display_flush(state->display) == -1 && errno == EAGAIN) {
    // This space is intentionally left blank
}
if (errno != 0) {
     break;
}
```

Reply:
I thought the documentation meant you can't do that, you have to `poll` before you can `flush` again.

Reply:
Oh, sorry, you're right. My bad.

Reply:
However we never wait for the FD to become writable.

Reply:
Isn't that what `poll` does?

Reply:
`poll` lets you know whether an FD is readable and/or writable. When reading we make sure that the FD is readable. However we don't check if it's writable before writing.